bot/exts/recruitment/talentpool/_cog.py

```diff
@@ -338,6 +348,97 @@ async def nominate_command(self, ctx: Context, user: MemberOrUser, *, reason: st
 
         await self._nominate_user(ctx, user, reason)
 
+    @app_commands.checks.has_any_role(*STAFF_ROLES)
+    async def _nominate_context_callback(self, interaction: discord.Interaction, message: discord.Message) -> None:
+        """Create or update a nomination for the author of the selected message."""
+        if message.author.bot:
+            await interaction.response.send_message(
+                ":x: I'm afraid I can't do that. Only humans can be nominated.",
+                ephemeral=True
+            )
+            return
+
+        if isinstance(message.author, Member) and any(role.id in STAFF_ROLES for role in message.author.roles):
+            await interaction.response.send_message(
+                ":x: Nominating staff members, eh? Here's a cookie :cookie:",
+                ephemeral=True
+            )
+            return
+
+        # Search for an existing nomination, if we find one we will append the message onto the existing reason.
+        maybe_nom = await self.api.get_nomination_reason(message.author.id, interaction.user.id)
+
+        if maybe_nom:
+            nom, reason = maybe_nom
+
+            new_reason = f"""
+            {reason}
+
+            {message.jump_url}
+            """
+
+            await self.api.edit_nomination_entry(nom.id, actor_id=interaction.user.id, reason=new_reason)
+
+            await interaction.response.send_message(
+                ":white_check_mark: Appended message to your existing nomination",
+                ephemeral=True
+            )
+        else:
+            # If no nomination is found, we will create a new one with the selected message
+            reason = f"Nominated from: {message.jump_url}"
+            try:
+                await self.api.post_nomination(message.author.id, interaction.user.id, reason)
+
+                await interaction.response.send_message(
+                    f":white_check_mark: The nomination for {message.author.mention} has been added to the talent pool",
+                    ephemeral=True
+                )
+            except ResponseCodeError as e:
+                match (e.status, e.response_json):
+                    case (400, {"user": _}):
+                        await interaction.response.send_message(
+                            f":x: {message.author.mention} can't be found in the database tables.",
```

Comment:
Is this something we should raise a sentry warning about, since it I don't think it's somthing that shiould acutally be possible

Reply:
It's copied from the other error handling, but I'll add a warning here

Reply:
I don't think we need one on the other one as the messages are not ephemeral